client/pages/NewTrip/NewTrip.jsx

```diff
@@ -100,30 +107,45 @@ class NewTrip extends Component {
   }
 
   getPointName = (startPoint) => {
-    const { google } = this.props;
     const name = new Promise((resolve, reject) => {
-      if (!google) {
+      if (!window.google) {
         reject(new Error('API is undefined'));
       }
-      const geocoder = new google.maps.Geocoder();
+      const geocoder = new window.google.maps.Geocoder();
       geocoder.geocode({ location: startPoint }, res => resolve(
         this.getParsetNameFromGeocode(res[0])
       ));
     });
     return name;
   }
 
-  addMarkers = (latLng) => {
+  addMarkers = (latLng, pointFlag) => {
     const { markers } = this.state;
-    if (markers.length > 1) {
-      this.setState(prevState => prevState.markers.pop());
+    let flag = markers.length ? 'B' : 'A';
```

Comment:
make a const for A and B

Reply:
very hard to understand this ternary. Please add comment and use explicit comparison for length